Simplenote/src/main/java/com/automattic/simplenote/AddTagActivity.kt

```diff
@@ -0,0 +1,130 @@
+package com.automattic.simplenote
+
+import android.os.Bundle
+import android.os.Handler
+import android.text.method.LinkMovementMethod
+import android.util.TypedValue
+import android.view.ContextThemeWrapper
+import android.view.View
+import android.widget.TextView
+import androidx.activity.viewModels
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.widget.doAfterTextChanged
+import com.automattic.simplenote.databinding.ActivityTagAddBinding
+import com.automattic.simplenote.utils.DisplayUtils
+import com.automattic.simplenote.utils.HtmlCompat
+import com.automattic.simplenote.utils.ThemeUtils
+import com.automattic.simplenote.viewmodels.AddTagViewModel
+import com.automattic.simplenote.widgets.MorphCircleToRectangle
+import com.automattic.simplenote.widgets.MorphSetup
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class AddTagActivity : AppCompatActivity() {
+    private val viewModel: AddTagViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        ThemeUtils.setTheme(this)
+        super.onCreate(savedInstanceState)
+        val binding: ActivityTagAddBinding = ActivityTagAddBinding.inflate(layoutInflater)
+
+        binding.setObservers()
+        binding.setupLayout()
+        binding.setupViews()
```

Comment:
Praise (❤️): I love what you did with the binding extension functions and using this pattern to build the UI, we are doing the same on `WordPress` so it is great to see it happening on `Simplenote` as well, kudos!

PS: Same applies to all the other places you did the same!

Reply:
Yes, I saw that in some files we discussed before and I thought that is a great way to reduce boilerplate and reduce the line length.